### ISSUE TITLE
[Feature Req]: Support SELECT queries in templating query

### ISSUE THREAD

**Issue by ab77** [grafana/grafana#5013]
Thank you very much for making this product available (and for free!).

I am having an issue with template variables since upgrading to `v3.0.2` (from `v.3.0.0.-pre1`).

`grafana_3.0.2-1463058303_armhf` with `influxdb_0.12.1-1_armhf.deb` running on `Linux 4.1.21-v7+ #872 SMP armv7l GNU/Linux`.
- What did you do?
  Added template variable to a dashboard using query `SELECT last("value") FROM "geo"` with regex extraction `([0-9\.]+,[0-9\.\-]+)`.

The actual query directly from InfluxDB returns:

```
> SELECT last("value") FROM "geo"
name: geo
---------
time    last
0       52.439034333,-0.1250985
```

The HTTP response from Grafana running this query is:

```
{"results":[{"series":[{"name":"geo","columns":["time","last"],"values":[[0,"52.439021333,-0.125126833"]]}]}]}
```
- What was the expected result?
  The expected result (was working with `v3.0.0-pre1`) was the displaying of the value as the template variable (e.g. `52.439034333,-0.1250985`).
- What happened instead?
  Instead, got `None` as the template variable. If I change the extraction to `(.*)`, the template variable is set to `0`, which is presumably the value in the `time` column.

I suspect Grafana is grabbing the value from the first `time` column or I am doing something wrong...

Any ideas?

Cheers!

**Comment by torkelo** [comment]
The InfluxDB data source in grafana only support SHOW style queries in templating, not metric queries (SELECT)

**Comment by ab77** [comment]
OK, thank you. I totally missed that, although I do remember seeing in in the docs now.

**Comment by ab77** [comment]
In the mean time, is there any way to pass a value in a singlestat panel to an external web link?

For example, if I have a singlestat panel displaying value "foo", I'd like the external link to be constructed with "foo" being passed as a query string parameter, as follows:
https://www.google.co.uk/maps?q=foo

I worked out how to do this with template vars in pre 3.0.2, but perhaps there is another more elegant way?

**Comment by burdandrei** [comment]
+1 for that

**Comment by erSitzt** [comment]
I'm not sure if my problem is related to this issue.

I have a template query to get all fields of a measurement 

`$metrics` with this query `SHOW FIELD KEYS from "measurementname"`

The resulting query to influxdb looks like this:

`q:SELECT "{metrics.berechtigung_sqlrequest,metrics.einstellung_rethink_connect_db,metrics.favorit_rethink_connect_db,metrics.favorit_rethink_delete}" FROM /demo/ WHERE time > now() - 7d`

or

`q:SELECT "/^{loadtime,resizetime,totaltime}$/" FROM "portal/imageproxy/master" WHERE time > now() - 7d`

This would work if the default value for a template variable would not be `field (/^$metrics$/)` and the list of values from the template query would not be wrapped in quotation marks  and curly braces.

This is a working version of the same query as above 

`SELECT metrics.berechtigung_sqlrequest,metrics.einstellung_rethink_connect_db,metrics.favorit_rethink_connect_db,metrics.favorit_rethink_delete FROM /demo/ WHERE time > now() - 7d`

If influxdb does not support regex in SELECT statements, is it possible to change the behaviour on template variable expansion in the SELECT part ?

regards 
Dennis

**Comment by conet** [comment]
I would use this feature to obtain template values from a TOP query, is there another way to do that?

**Comment by burdandrei** [comment]
@conet  everyone want to use it this way ;)

**Comment by conet** [comment]
@burdandrei Then I would like to share the workaround I'm using now (since it took me more than I expected to figure it out, others might find this useful). 

With kapacitor I'm running the top query periodically and I'm tagging all topN results (with a 'topN' tag) in another measurement. Then I'm using the [Multi-Select Tagging](http://docs.grafana.org/reference/templating/#multi-select) feature to easily select tag values standing for the topN values in the measurement.

**Comment by burdandrei** [comment]
Thanks @conet!
I wanted to do something like this, but was to lasy for this, also you can do it with continous query ;)

**Comment by Sebo-90** [comment]
+1

**Comment by omarmarquez** [comment]
+1

**Comment by jenningsloy318** [comment]
+1

**Comment by kb1lqc** [comment]
Any suggestions on formatting a tag to hold multiple text values for templating? I assume I can use regex to parse between deliminators and make one tag per unique entry? Basically I want to use templating queries to pull chart names from the database. So one entry would have something like "chart1,chart2,chart3" and then I could label each grafana chart with each name.

**Comment by RicardoSantosDeveloper** [comment]
+1

**Comment by ricsil** [comment]
+1, very much needed

**Comment by torkelo** [comment]
implemented by https://github.com/grafana/grafana/pull/10677

**Comment by daniellee** [comment]
Had to revert #10677 so reopening this issue.

**Comment by hanynowsky** [comment]
Excellent news. That was a blocker feature.

**Comment by serputko** [comment]
Hi guys,
is it possible now to get timestamp with SELECT query in templating as it was in previous grafana versions?

**Comment by hanynowsky** [comment]
Hello, guys did this get on 5.2.2 ?

**Comment by marefr** [comment]
@hanynowsky included in v5.2

**Comment by fredduwez** [comment]
Great feature! I may be wrong, but I can't find the documentation about it (http://docs.grafana.org/features/datasources/influxdb/#templating) and I would like to know if and how I can use the $_interval into the templating select query.

**Comment by fredduwez** [comment]
Update: the most important variable I tried to use in my template query is in fact $timeFilter (I am trying to list all the active series by their tag value for a given period), and I don't know how to make it work for the period.

Ok request:
SELECT campaignID, v FROM (SELECT sum("value") as v FROM "api.prediction.prescribed.counter" WHERE ("stage" =~ /^$Stage$/ AND "datastreamID" =~ /^$Stream$/) GROUP BY time(10000d), "campaignID")

Broken request (only in template query space):
SELECT campaignID, v FROM (SELECT sum("value") as v FROM "api.prediction.prescribed.counter" WHERE ("stage" =~ /^$Stage$/ AND "datastreamID" =~ /^$Stream$/) **AND $timeFilter** GROUP BY time(10000d), "campaignID")

**Comment by marefr** [comment]
@fredduwez this issue in only for supporting selection of field values in template variables. #12598 probably adds support for what you want and it will be included in Grafana v5.3 (not yet released). You can try out the nightly build or the v5.3 beta (out soon hopefully).

**Comment by kantsuw** [comment]
@marefr 

As I checked in 5.2.3 release , #12598 already merged into it.

However as test with $timeFilter in query variable it not worked.

Templating
Template variables could not be initialized: error parsing query: missing parameter: timeFilter

 So I have to wait on 5.3 for this feature right ?
